sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -2112,6 +2114,477 @@ internal virtual void SetNameFieldsIfNull()
+                        int directoriesSuccessfulCount = 0;
+                        int filesSuccessfulCount = 0;
+                        int failureCount = 0;
+                        int batchesCount = 0;
+                        do
+                        {
+                            jsonResponse =
+                                await DataLakeRestClient.Path.SetAccessControlRecursiveAsync(
+                                    clientDiagnostics: ClientDiagnostics,
+                                    pipeline: Pipeline,
+                                    resourceUri: DfsUri,
+                                    mode: mode,
+                                    maxRecords: options.BatchSize,
+                                    version: Version.ToVersionString(),
+                                    acl: accessControlList,
+                                    async: async,
+                                    continuation: continuationToken,
+                                    forceFlag: options.ContinueOnFailure ? true : (bool?)default,
+                                    cancellationToken: cancellationToken)
+                                .ConfigureAwait(false);
+
+                            continuationToken = jsonResponse.Value.Continuation;
+                            if (!string.IsNullOrEmpty(continuationToken))
+                            {
+                                lastContinuationToken = continuationToken;
+                            }
+                            using (var document = JsonDocument.Parse(jsonResponse.Value.Body))
+                            {
+                                var response = document.RootElement.DeserializeSetAccessControlRecursiveResponse();
+                                int currentDirectoriesSuccessfulCount = response.DirectoriesSuccessful ?? 0;
+                                int currentFilesSuccessfulCount = response.FilesSuccessful ?? 0;
+                                int currentFailureCount = response.FailureCount ?? 0;
+                                directoriesSuccessfulCount += currentDirectoriesSuccessfulCount;
+                                filesSuccessfulCount += currentFilesSuccessfulCount;
+                                failureCount += currentFailureCount;
+                                if (progressHandler != null)
+                                {
+                                    var failedEntries = response.FailedEntries
+                                        .Select(failedEntry => new AccessControlChangeFailure()
+                                        {
+                                            Name = failedEntry.Name,
+                                            IsDirectory = failedEntry.Type.Equals("DIRECTORY", StringComparison.InvariantCultureIgnoreCase),
+                                            ErrorMessage = failedEntry.ErrorMessage,
+                                        }).ToList();
```

Comment:
@kasobol-msft, @amnguye

Reply:
In particular, what if we add "the last batch of failures" to the result?  Then you wouldn't need IProgress notifications on the happy path because you could check for failures, print out what broke, and print out a continuation token that could be used to resume after fixing.  We'd only hold on to the last batch so it wouldn't be an upsetting amount of data in the event you passed the force flag.  This would also let us move the IProgress handler into the Options type and simplify the signature.

Reply:
that might confuse some if continueOnFailure = true. In that case IProgress would be used to catch all failed entries and "the last batch of failures" in the summary result might not contain all data.
Btw. this way we'd make "batching" concept more visible which may lead customers to explore advanced options prematurely.

Reply:
I care less about confusing users of advanced scenarios and making customers on the happy path successful.  Having to provide an IProgress notification handler to see what went wrong feels like we're not making that easy enough.  I don't think anyone will get the IProgress part correct without reading a sample first.  If we put some of the errors in the result, I think people would have a way higher chance of writing correct code.

Reply:
I'd consider cases with continueOnFailure=false and continueOnFailure=true two basic scenarios (conversely the later might be quite popular choice). Cases where someone tries to break down iteration seem to be more advanced.

Reply:
after giving it more thoughts I think it's not a bad idea to expose last seen failures in the overall result as long as we documentation of that field is going to mention the limitation and direct to IProgress for capturing full set.

Reply:
Let's noodle on it and see if there's a clean way of making it happen.  Maybe writing sample code is a good place to start.  We can do it outside this PR of course.